examples/chip-tool/main.cpp

```diff
@@ -156,9 +157,12 @@ void HandleDataModelMessage(System::PacketBuffer * buffer)
 
 void OnMessage(DeviceController::ChipDeviceController * deviceController, void * appReqState, System::PacketBuffer * buffer)
 {
-    size_t data_len    = buffer->DataLength();
-    waitingForResponse = false;
-    cvWaitingForResponse.notify_all();
+    size_t data_len = buffer->DataLength();
+    {
+        std::lock_guard<std::mutex> lk(cvWaitingForResponseMutex);
+        waitingForResponse = false;
+        cvWaitingForResponse.notify_all();
```

Comment:
This needs to happen after the `lock_guard` goes out of scope and the lock is unlocked, since whoever wakes up will try to acquire the lock.